src/ucs/sys/netlink.c

```diff
@@ -131,33 +132,42 @@ ucs_netlink_send_request(int protocol, unsigned short nlmsg_type,
     }
 
     /* get message size */
-    status = ucs_socket_recv_nb(netlink_fd, NULL, MSG_PEEK | MSG_TRUNC,
-                                &recv_msg_len);
-    if (status != UCS_OK) {
-        ucs_error("failed to get netlink message size %d (%s)",
-                  status, ucs_status_string(status));
-        goto out;
-    }
+    do {
+        size_t recv_msg_len = 0;
+        char *recv_msg;
```

Comment:
can we move the definitions back to be beginning of the function? to align code style